# HG changeset patch
# User Rik <rik@octave.org>
# Date 1463371710 25200
#      Sun May 15 21:08:30 2016 -0700
# Node ID 764e586dca4b25ab640bd5f6eb3d72e61e79871d
# Parent  ac59b72712fd9af86afb947be735c2b06b14eb38
Avoid use of REJECT in parser (bug #45947).

* lex.ll (HANDLE_IDENTIFIER): Strip get/set from input if pattern
(get|set)\s*\( matched.
* lex.ll: Change pattern to "(get|set)\s*\(" from "(get|set)/\s*\("
to eliminate trailing context usage which can invoke REJECT.

diff --git a/libinterp/parse-tree/lex.ll b/libinterp/parse-tree/lex.ll
--- a/libinterp/parse-tree/lex.ll
+++ b/libinterp/parse-tree/lex.ll
@@ -312,17 +312,20 @@ object) relevant global values before an
               && curr_lexer->previous_token_may_be_command ()) \
             { \
               yyless (0); \
               curr_lexer->push_start_state (COMMAND_START); \
             } \
           else \
             { \
               if (get_set) \
-                curr_lexer->maybe_classdef_get_set_method = false; \
+                { \
+                  yyless (3); \
+                  curr_lexer->maybe_classdef_get_set_method = false; \
+                } \
  \
               int id_tok = curr_lexer->handle_identifier (); \
  \
               if (id_tok >= 0) \
                 return curr_lexer->count_token_internal (id_tok); \
             } \
         } \
     } \
@@ -1249,17 +1252,17 @@ ANY_INCLUDING_NL (.|{NL})
 
 %{
 // Identifiers.
 
 // Don't allow get and set to be recognized as keywords if they are
 // followed by "(".
 %}
 
-(set|get)/{S}*\( {
+(set|get){S}*\( {
     HANDLE_IDENTIFIER ("(set|get)/{S}*\\(", true);
   }
 
 {IDENT} {
     HANDLE_IDENTIFIER ("{IDENT}", false);
   }
 
 %{
